Tests/PackageModelSyntaxTests/ManifestEditTests.swift

```diff
@@ -0,0 +1,312 @@
+            let package = Package(
+                name: "packages",
+                dependencies: [ ]
+            )
+            """,
+            expectedManifest: """
+            let package = Package(
+                name: "packages",
+                dependencies: [
+                    .package(url: "https://github.com/apple/swift-system.git", .branch("main")),
+                ]
+            )
+            """) { manifest in
+            AddPackageDependency.addPackageDependency(
+                Self.swiftSystemPackageDependency,
+                to: manifest,
+                manifestDirectory: try! AbsolutePath(validating: "/")
+            )
+        }
+    }
+
+    func testAddPackageDependencyNoExistingAtEnd() {
+        assertManifestRefactor("""
+            let package = Package(
+                name: "packages"
+            )
+            """, 
+            expectedManifest: """
+            let package = Package(
+                name: "packages",
+                dependencies: [
+                    .package(url: "https://github.com/apple/swift-system.git", .branch("main")),
+                ]
+            )
+            """) { manifest in
+            AddPackageDependency.addPackageDependency(
+                Self.swiftSystemPackageDependency,
+                to: manifest,
+                manifestDirectory: try! AbsolutePath(validating: "/")
+            )
+        }
+    }
+
+    func testAddPackageDependencyNoExistingMiddle() {
+        assertManifestRefactor("""
+            let package = Package(
+                name: "packages",
+                targets: []
+            )
+            """,
+            expectedManifest: """
+            let package = Package(
+                name: "packages",
+                dependencies: [
+                    .package(url: "https://github.com/apple/swift-system.git", .branch("main")),
+                ],
+                targets: []
+            )
+            """) { manifest in
+            AddPackageDependency.addPackageDependency(
+                Self.swiftSystemPackageDependency,
+                to: manifest,
+                manifestDirectory: try! AbsolutePath(validating: "/")
+            )
+        }
+    }
+}
+
+
+// FIXME: Copy-paste from _SwiftSyntaxTestSupport
+
+/// Asserts that the two strings are equal, providing Unix `diff`-style output if they are not.
+///
+/// - Parameters:
+///   - actual: The actual string.
+///   - expected: The expected string.
+///   - message: An optional description of the failure.
+///   - additionalInfo: Additional information about the failed test case that will be printed after the diff
+///   - file: The file in which failure occurred. Defaults to the file name of the test case in
+///     which this function was called.
+///   - line: The line number on which failure occurred. Defaults to the line number on which this
+///     function was called.
+public func assertStringsEqualWithDiff(
+  _ actual: String,
```

Comment:
These functions use formatting inconsistent with 4 spaces in the rest of the file.

Reply:
'cause I copy-pasted it, thanks!